packages/use-debounced-state/src/useDebouncedState.ts

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+
+export function useDebouncedState<T = unknown>(
+  defaultValue: T,
+  wait: number,
+  options = { leading: false },
+) {
+  const [value, setValue] = React.useState(defaultValue);
+  const timeoutRef = React.useRef<number>();
+  const leadingRef = React.useRef(true);
+
+  const clearTimeout = () => void window.clearTimeout(timeoutRef.current);
```

Comment:
do you think this would benefit from being wrapped in a `useCallback` since it's called in a `useEffect`?